Tests/MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -343,6 +343,66 @@ class RouteControllerTests: TestCase {
+        RouteControllerProactiveReroutingInterval = 1
+        
+        let coordinates = [
+            CLLocationCoordinate2D(latitude: 38.853108, longitude: -77.043331),
+            CLLocationCoordinate2D(latitude: 38.910736, longitude: -76.966906),
+        ]
+        
+        let options = NavigationRouteOptions(coordinates: coordinates)
+        let route = Fixture.route(from: "DCA-Arboretum", options: options)
+        let replayLocations = Fixture.generateTrace(for: route).shiftedToPresent().qualified()
+        let routeResponse = RouteResponse(httpResponse: nil,
+                                          routes: [route],
+                                          options: .route(.init(locations: replayLocations,
+                                                                profileIdentifier: nil)),
+                                          credentials: .mocked)
+        
+        let indexedRouteResponse = IndexedRouteResponse(routeResponse: routeResponse,
+                                                        routeIndex: 0)
+        
+        let routingProviderStub = CustomRoutingProviderStub()
+        
+        routingProviderStub.indexedRouteStub = { completion in
+            let route = Fixture.route(from: "DCA-Arboretum-duration-edited", options: options)
+            let replayLocations = Fixture.generateTrace(for: route).shiftedToPresent().qualified()
+            let routeResponse = RouteResponse(httpResponse: nil,
+                                              routes: [route],
+                                              options: .route(.init(locations: replayLocations,
+                                                                    profileIdentifier: nil)),
+                                              credentials: .mocked)
+            
+            completion(.success(IndexedRouteResponse(routeResponse: routeResponse,
+                                                     routeIndex: 0)))
+        }
+        
+        let routeController = RouteController(indexedRouteResponse: indexedRouteResponse,
+                                              customRoutingProvider: routingProviderStub,
+                                              dataSource: self)
+        let routerDelegateSpy = RouterDelegateSpy()
+        let routeExpectation = XCTestExpectation(description: "Proactive ReRoute should be called")
+        
+        routerDelegateSpy.onShouldProactivelyRerouteFrom = { _, _ in
+            routeExpectation.fulfill()
+            return true
+        }
+        
+        routeController.delegate = routerDelegateSpy
+        
+        let locationManager = ReplayLocationManager(locations: [CLLocation(coordinate: coordinates[0])].shiftedToPresent())
+        locationManager.startDate = Date()
+        locationManager.delegate = routeController
+        
+        locationManager.speedMultiplier = 1
+        locationManager.startUpdatingLocation()
+        wait(for: [routeExpectation], timeout: 15)
```

Comment:
Do we need such a large timeout?

Reply:
It takes some time for this mechanism to trigger (init everything, wait for location manager to provide an update, wait for NN status, checks...). In result, this test takes around 9s to complete, even though `RouteControllerProactiveReroutingInterval` is set to 1 sec.